logging/test/mock_hub.dart

```diff
@@ -10,24 +10,32 @@ class MockHub with NoSuchMethodProvider implements Hub {
   final List<Breadcrumb> breadcrumbs = [];
   final List<CapturedEvents> events = [];
   final _options = SentryOptions(dsn: 'fixture-dsn');
+  final List<Hint> breadcrumbHints = [];
+  final List<Hint> eventHints = [];
 
   @override
   @internal
   SentryOptions get options => _options;
 
   @override
-  Future<void> addBreadcrumb(Breadcrumb crumb, {dynamic hint}) async {
+  Future<void> addBreadcrumb(Breadcrumb crumb, {Hint? hint}) async {
     breadcrumbs.add(crumb);
```

Comment:
Crumbs here should be likely similar to `CapturedEvents` or a Map, otherwise this will cause flakiness, the index will be wrong if you capture a crumb with and without a hint.